TrafficSimulator/Assets/Prefabs/Car/AutoDrive.cs

```diff
@@ -268,12 +320,22 @@ private void Q_UpdateBrakeTarget()
         {
             // Set the brake target point to the point closest to the target that is at least _brakeDistance points away
             // If the road end behaviour is set to stop and the brake target is the end node, do not update the brake target
-            while (Vector3.Distance(transform.position, _brakeTarget.Position) < _brakeDistance && (_brakeTarget != _endNode || _roadEndBehaviour == RoadEndBehaviour.Loop))
+            // If the next node has a vehicle, do not update the brake target
+            while (updateBrakeTarget())
             {
-                _brakeTarget = GetNextLaneNode(_brakeTarget, 0, _roadEndBehaviour == RoadEndBehaviour.Loop);
+                _brakeTarget = GetNextLaneNode(_brakeTarget, 0, true);
             }
         }
 
+        private bool updateBrakeTarget()
+        {
+            bool _nextNodeHasVehicle = GetNextLaneNode(_brakeTarget, 0, true).HasVehicle();
+            bool _brakeTargetIsEndNode = _brakeTarget.Type == RoadNodeType.End && _brakeTarget.Position != _startNode.Position;
+            bool _brakeTargetIsEndNodeAndLoop = _brakeTargetIsEndNode && _roadEndBehaviour == RoadEndBehaviour.Loop;
+            bool _brakeDistanceIsGreaterThanBrakeTargetDistance = Vector3.Distance(transform.position, _brakeTarget.Position) < _brakeDistance;
+            return _brakeDistanceIsGreaterThanBrakeTargetDistance && !_nextNodeHasVehicle && (!_brakeTargetIsEndNode | _brakeTargetIsEndNodeAndLoop);
```

Comment:
Is this really supposed to be the logical operator `|` instead of the conditional logical operator `||`? The latter uses lazy evaluation which seems to be the only difference but would be better in this case unless I am missing something. 

Either way it either needs to be changed to the `||` operator, or a comment needs to be provided explaining exactly why that could not be used